src/fidesops/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -0,0 +1,276 @@
+
+
+@router.post(
+    CONSENT_REQUEST,
+    status_code=HTTP_200_OK,
+    response_model=ConsentRequestResponse,
+)
+def create_consent_request(
+    *,
+    db: Session = Depends(get_db),
+    data: Identity,
+) -> ConsentRequestResponse:
+    """Creates a verification code for the user to verify access to manange consent preferences."""
+    if not config.redis.enabled:
+        raise FunctionalityNotConfigured(
+            "Application redis cache required, but it is currently disabled! Please update your application configuration to enable integration with a redis cache."
+        )
+
+    if not config.execution.subject_identity_verification_required:
+        raise FunctionalityNotConfigured(
+            "Subject identity verification is required, but it is currently disabled! Please update your application configuration to enable subject identity verification."
+        )
+
+    if not data.email:
+        raise HTTPException(HTTP_400_BAD_REQUEST, detail="An email address is required")
+
+    identity = ProvidedIdentity.filter(
+        db=db,
+        conditions=(
+            (ProvidedIdentity.field_name == ProvidedIdentityType.email)
+            & (ProvidedIdentity.hashed_value == ProvidedIdentity.hash_value(data.email))
+        ),
+    ).first()
+
+    if not identity:
+        provided_identity_data = {
+            "privacy_request_id": None,
+            "field_name": "email",
+            "encrypted_value": {"value": "test@email.com"},
+        }
+        identity = ProvidedIdentity.create(db, data=provided_identity_data)
+
+    consent_request_data = {
+        "provided_identity_id": identity.id,
+    }
+    consent_request = ConsentRequest.create(db, data=consent_request_data)
+    verificaiton_code = _send_verification_code_to_user(db, consent_request, data.email)
+    return ConsentRequestResponse(identity=data, verification_code=verificaiton_code)
+
+
+@router.post(
+    CONSENT_REQUEST_VERIFY,
+    status_code=HTTP_200_OK,
+    response_model=ConsentPreferences,
+)
+def consent_request_verify(
+    *,
+    consent_request_id: str,
+    db: Session = Depends(get_db),
+    data: VerificationCode,
+) -> ConsentPreferences:
+    """Verifies the verification code and returns the current consent preferences if successful."""
+    provided_identity = _get_consent_request_and_provided_identity(
+        db=db, consent_request_id=consent_request_id, verification_code=data.code
+    )
+
+    if not provided_identity.encrypted_value:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND, detail="Provided identity missing email"
+        )
+
+    consent = Consent.filter(
+        db=db, conditions=Consent.provided_identity_id == provided_identity.id
+    ).all()
+
+    identity = (
+        Identity(email=provided_identity.encrypted_value["value"])
+        if provided_identity.field_name == ProvidedIdentityType.email
+        else Identity(phone_number=provided_identity.encrypted_value["value"])
```

Comment:
provided_identities created through this workflow only have a phone_number on this so far.  Is this for a provided_identity created elsewhere through the privacy request workflow?

Reply:
Currently only email is supported, but I am assuming in the future phone number will work also so this was in preparation for that. Since the front end will control what is sent I'm assuming only emails will be sent for now. Maybe a bad assumption?

Reply:
OK I think that assumption makes sense